catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/activity/ProjectUploadActivity.java

```diff
@@ -185,6 +187,7 @@ private void onNextButtonClick() {
 		ProjectManager projectManager = ProjectManager.getInstance();
 
 		if (Utils.isDefaultProject(projectManager.getCurrentProject(), this)) {
+			ToastUtil.showError(this, R.string.error_upload_default_project);
```

Comment:
Didn't you want to check this before? i.e. in the `Activity` that started the `UploadActivity`? so e.g. in `MainMenuFragment` in `public void onItemClick(@ButtonId int id) {..}` in the `case UPLOAD:` and wherever else the activity should be started.
```
void someMethod() {
   if (Utils.isDefaultProject(projectManager.getCurrentProject(), this)) {
       ToastUtil.showError(this, R.string.error_upload_default_project);
   } else {
        Intent intent = new Intent(getActivity(), ProjectUploadActivity.class)
                 .putExtra(ProjectUploadActivity.PROJECT_NAME, Utils.getCurrentProjectName(getActivity()));
        startActivity(intent);
```

And please also add a test for that behaviour then. :wink: